internal/provider/pod.go

```diff
@@ -406,32 +406,37 @@ func unitPrefix(namespace, podName string) string {
 	return prefix + separator + namespace + separator + podName
 }
 
-func Image(name string) string {
-	el := strings.Split(name, separator) // assume well formed
+// Name returns <namespace>.<podname> from a well formed name.
+// Units are named as 'systemk.<namespace>.<podname>.<image> .
+func Name(name string) string {
+	el := strings.Split(name, separator)
 	if len(el) < 4 {
 		return ""
 	}
-	return el[3]
+	return el[1] + separator + el[2]
 }
 
-func Name(name string) string {
-	el := strings.Split(name, separator) // assume well formed
+// Image returns the <image> from the well formed name. See Name.
+func Image(name string) string {
```

Comment:
Same comment as above regarding my preference for multiple arguments vs an opaque one and string splitting.